### ISSUE TITLE
Detect assignments to the the blank identifier

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#869]
I feel like assigning a variable to the blank identifier is a code smell and in most cases a left over from the development / debugging process to silent the compiler.

Therefore I suggest a checker to detect cases, where a variable is assigned to the blank identifier.

Before:

```Go
x := foo()
_ = x
```

After:

```Go
x := foo()
```

In a slightly less aggressive variant, the check could only fail, if the assignment to the blank identifier looks like a left over.

Before:

```Go
x := foo()
_ = x
return x
```
After:

```Go
x := foo()
return x
```

Additionally, [Effective Go](https://golang.org/doc/effective_go.html#blank_unused) also mentions the usage of the blank identifier to prevent compiler complaints by assigning a symbol of the otherwise unused package to the blank identifier. This could be detected as well.

**Comment by quasilyte** [comment]
As with #873, it can be easily expressed with AST rules:
https://github.com/quasilyte/go-ruleguard/commit/cc95a0e15f7d0f18785eb878eb683801ea846db4

```go
m.Match(`_ = $v`).
		Where(m["v"].Pure).
		Report(`please remove the assignment to _`)
```

**Comment by cristaloleg** [comment]
Closing as resolved in ruleguard.
